src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, ElementRef, Inject, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { faCircle, faExpand, faPaperPlane, faXmark } from '@fortawesome/free-solid-svg-icons';
+import { MAT_DIALOG_DATA, MatDialog } from '@angular/material/dialog';
+import { IrisStateStore } from 'app/iris/state-store.service';
+import { ConversationErrorOccurredAction, StudentMessageSentAction } from 'app/iris/state-store.model';
+import { IrisHttpMessageService } from 'app/iris/http-message.service';
+import { IrisClientMessage, IrisMessage, IrisSender } from 'app/entities/iris/iris-message.model';
+import { IrisMessageContent, IrisMessageContentType } from 'app/entities/iris/iris-content-type.model';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'jhi-exercise-chat-widget',
+    templateUrl: './exercise-chat-widget.component.html',
+    styleUrls: ['./exercise-chat-widget.component.scss'],
+})
+export class ExerciseChatWidgetComponent implements OnInit, OnDestroy {
+    @ViewChild('chatWidget') chatWidget!: ElementRef;
+    @ViewChild('chatBody') chatBody!: ElementRef;
```

Comment:
I haven't seen exclamation marks in such a place. What do they do here?

Reply:
this one was added in this PR https://github.com/ls1intum/Artemis/pull/6584. Would you pls explain this @eceeeren @asliayk ?

Reply:
An exclamation mark means a field will never be undefined/null. Since the corresponding divs are always present, a stronger type makes sense.